maven-model-builder/src/main/java/org/apache/maven/utils/Os.java

```diff
@@ -0,0 +1,383 @@
+/*
```

Comment:
This can't be copied over as is. It needs to be cleaned up and rethought. It's horribly out of date.

Reply:
This class is used in two cases: for profile activation and to check if the OS is windows.  The second case is still correct and for the first one, I don't think anything is broken.  So yes, we may not be able to detect if we're on OSX Big Sur, Monterey or whatever, but we don't need it. It solves our needs, I think it's fine.

Reply:
It's public, not just an implementation detail, so I think more care is warranted. If it can be made non-public it can be cleaned up later, but we should be cautious about locking in APIs we don't need and don't want to support long term.

Reply:
The values might be used but I don't think these constants are. enforcer-rules has its own variant of this class. I'm thinking about all of these but anything we make public here makes it hard to fix or improve any of this later.